ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLTask.java

```diff
@@ -82,8 +89,32 @@ public int execute() {
         throw new IllegalArgumentException("Unknown DDL request: " + ddlDesc.getClass());
       }
     } catch (Throwable e) {
+      LOG.error("DDLTask failed", e);
+      int errorCode = ErrorMsg.getErrorMsg(e.getMessage()).getErrorCode();
+      try {
+        ReplicationMetricCollector metricCollector = work.getMetricCollector();
+        if (errorCode > 40000) {
+          //in case of replication related task, dumpDirectory should not be null
+          if(work.dumpDirectory != null) {
+            Path nonRecoverableMarker = new Path(work.dumpDirectory, ReplAck.NON_RECOVERABLE_MARKER.toString());
+            org.apache.hadoop.hive.ql.parse.repl.dump.Utils.writeStackTrace(e, nonRecoverableMarker, conf);
+            if(metricCollector != null){
+              metricCollector.reportStageEnd(getName(), Status.FAILED_ADMIN, nonRecoverableMarker.toString());
+            }
+          }
+          if(metricCollector != null){
```

Comment:
this is needed only in replication case

Reply:
In replication flows, dumpDirectory and metricCollector both should be non-null. This line considers the corner case where metricCollector might have been configured but not dumpDirectory. Still it is a replication case since only replication tasks can initialise and pass metricCollector. So we should indicate FAILED_ADMIN state at-least (non-recoverable path is null).